### PR TITLE
Fix: Nullable input params for setController and setTemplate

### LINKED ISSUE #14048: [Bug]: Document Types cannot save type with empty Controller or Template
### Expected behavior

Adding or updating a document type should be possible without error message if the Controller or Template is empty.

### Actual behavior

Updating or adding a new document type will result in an error message. 

Empty controller
image

Empty template
image

### Steps to reproduce

Go to Settings > Document Types and try to add a new Document Type or try updating (e.g. change priority) an existing one.

### PR DESCRIPTION
**Pull request by mattamon**
## Changes in this pull request  
Resolves #14048 

## Additional info  
`setController` and `setTemplate` now take null as an argument.
property group is now nullable too

### CODE REVIEW

File: models/Document/DocType.php [hunk 1]
```diff
@@ -52,7 +52,7 @@ class DocType extends Model\AbstractModel
      *
      * @var string
```
Comment (blankse) [inline models/Document/DocType.php]:
```suggestion
```

File: models/Document/DocType.php [hunk 2]
```diff
@@ -168,7 +168,7 @@ public function getTemplate(): ?string
         return $this->template;
     }
 
-    public function setController(string $controller): static
+    public function setController(?string $controller): static
```
Comment (blankse) [inline models/Document/DocType.php:150]:
```suggestion
    /**
     * @return $this
     */
    public function setController(?string $controller): static
```

File: models/Document/DocType.php [hunk 3]
```diff
@@ -196,7 +196,7 @@ public function setGroup(string $group): static
         return $this;
     }
 
-    public function setTemplate(string $template): static
+    public function setTemplate(?string $template): static
```
Comment (blankse) [inline models/Document/DocType.php:190]:
```suggestion
    /**
     * @return $this
     */
    public function setTemplate(?string $template): static
```

File: models/Document/DocType.php (hunk 1 above)
Comment (mattamon) [reply models/Document/DocType.php]:
```suggestion
     * @var string|null
```
I'd rather go for string|null to keep it consistent with the rest.

File: models/Document/DocType.php [hunk 4]
```diff
@@ -52,7 +52,7 @@ class DocType extends Model\AbstractModel
      *
      * @var string
      */
-    protected string $group;
+    protected ?string $group = null;
```
Comment (blankse) [inline models/Document/DocType.php:49]: The setGroup() method should also be accept null

File: models/Document/DocType.php (hunk 1 above)
Comment (blankse) [reply models/Document/DocType.php]: It shouldn't be redundant php docs. You can remove it also from the rest.

**Review by blankse — APPROVED** [review]
LGTM 👍🏼

**Review by aashan10 — APPROVED** [review]

**Pull request merged**